pkg/resources/api_authentication_integration_common.go

```diff
@@ -0,0 +1,356 @@
+	},
+	"oauth_access_token_validity": {
+		Type:             schema.TypeInt,
+		Optional:         true,
+		ValidateFunc:     validation.IntAtLeast(0),
+		Default:          IntDefault,
+		Description:      "Specifies the default lifetime of the OAuth access token (in seconds) issued by an OAuth server.",
+		DiffSuppressFunc: IgnoreChangeToCurrentSnowflakeValueInDescribe("oauth_access_token_validity"),
+	},
+	"oauth_refresh_token_validity": {
+		Type:             schema.TypeInt,
+		Optional:         true,
+		ValidateFunc:     validation.IntAtLeast(1),
+		Description:      "Specifies the value to determine the validity of the refresh token obtained from the OAuth server.",
+		DiffSuppressFunc: IgnoreChangeToCurrentSnowflakeValueInDescribe("oauth_refresh_token_validity"),
+	},
+	"comment": {
+		Type:        schema.TypeString,
+		Optional:    true,
+		Description: "Specifies a comment for the integration.",
+	},
+	ShowOutputAttributeName: {
+		Type:        schema.TypeList,
+		Computed:    true,
+		Description: "Outputs the result of `SHOW SECURITY INTEGRATIONS` for the given security integration.",
+		Elem: &schema.Resource{
+			Schema: schemas.ShowSecurityIntegrationSchema,
+		},
+	},
+	DescribeOutputAttributeName: {
+		Type:        schema.TypeList,
+		Computed:    true,
+		Description: "Outputs the result of `DESCRIBE SECURITY INTEGRATIONS` for the given security integration.",
+		Elem: &schema.Resource{
+			Schema: schemas.DescribeApiAuthSecurityIntegrationSchema,
+		},
+	},
+}
+
+type commonApiAuthSet struct {
+	enabled                   *bool
+	oauthClientId             *string
+	oauthClientSecret         *string
+	comment                   *string
+	oauthAccessTokenValidity  *int
+	oauthClientAuthMethod     *sdk.ApiAuthenticationSecurityIntegrationOauthClientAuthMethodOption
+	oauthTokenEndpoint        *string
+	oauthRefreshTokenValidity *int
+}
+
+type commonApiAuthUnset struct {
+	comment *bool
+}
+
+func handleApiAuthUpdate(d *schema.ResourceData) (commonApiAuthSet, commonApiAuthUnset, error) {
+	set, unset := commonApiAuthSet{}, commonApiAuthUnset{}
+
+	if d.HasChange("enabled") {
+		// required field
+		set.enabled = sdk.Pointer(d.Get("enabled").(bool))
+	}
+
+	if d.HasChange("oauth_client_id") {
+		// required field
+		set.oauthClientId = sdk.Pointer(d.Get("oauth_client_id").(string))
+	}
+
+	if d.HasChange("oauth_client_secret") {
+		// required field
+		set.oauthClientSecret = sdk.Pointer(d.Get("oauth_client_secret").(string))
+	}
+
+	if d.HasChange("comment") {
+		if v, ok := d.GetOk("comment"); ok {
+			set.comment = sdk.Pointer(v.(string))
+		} else {
+			unset.comment = sdk.Pointer(true)
+		}
+	}
+
+	if d.HasChange("oauth_access_token_validity") {
+		if v := d.Get("oauth_access_token_validity").(int); v != IntDefault {
+			set.oauthAccessTokenValidity = sdk.Pointer(v)
+		} else {
+			// TODO(SNOW-1515781): use UNSET
+			set.oauthAccessTokenValidity = sdk.Pointer(0)
```

Comment:
Is 0 a Snowflake default for this value?

Reply:
Yes.